operator/builtin/input/aws/cloudwatch/cloudwatch.go

```diff
@@ -0,0 +1,372 @@
+		return nil, fmt.Errorf("invalid value for parameter 'start_at'")
+	}
+
+	var startAtEnd bool
+	switch c.StartAt {
+	case "beginning":
+		startAtEnd = false
+	case "end":
+		startAtEnd = true
+	default:
+		return nil, fmt.Errorf("invalid value '%s' for %s parameter 'start_at'", c.StartAt, operatorName)
+	}
+
+	cloudwatchInput := &CloudwatchInput{
+		InputOperator:       inputOperator,
+		logGroupName:        c.LogGroupName,
+		logStreamNames:      c.LogStreamNames,
+		logStreamNamePrefix: c.LogStreamNamePrefix,
+		region:              c.Region,
+		eventLimit:          c.EventLimit,
+		profile:             c.Profile,
+		pollInterval:        c.PollInterval,
+		startAtEnd:          startAtEnd,
+		persist: Persister{
+			DB: helper.NewScopedDBPersister(buildContext.Database, c.ID()),
+		},
+	}
+	return []operator.Operator{cloudwatchInput}, nil
+}
+
+// CloudwatchInput is an operator that reads input from AWS Cloudwatch Logs.
+type CloudwatchInput struct {
+	helper.InputOperator
+	cancel       context.CancelFunc
+	pollInterval helper.Duration
+
+	logGroupName        string
+	logStreamNames      []*string
+	logStreamNamePrefix string
+	region              string
+	eventLimit          int64
+	profile             string
+	startAtEnd          bool
+	startTime           int64
+	persist             Persister
+	wg                  sync.WaitGroup
+}
+
+// Start will start generating log entries.
+func (c *CloudwatchInput) Start() error {
+	ctx, cancel := context.WithCancel(context.Background())
+	c.cancel = cancel
+
+	if err := c.persist.DB.Load(); err != nil {
+		return err
+	}
+
+	return c.pollEvents(ctx)
+}
+
+// Stop will stop generating logs.
+func (c *CloudwatchInput) Stop() error {
+	c.cancel()
+	c.wg.Wait()
+	fmt.Printf("Closed all connections to Cloudwatch Logs")
+	return nil
+}
+
+// pollEvents gets events from AWS Cloudwatch Logs every poll interval.
+func (c *CloudwatchInput) pollEvents(ctx context.Context) error {
+	c.Infof("Started polling AWS Cloudwatch Logs group '%s' using poll interval of '%s'", c.logGroupName, c.pollInterval)
+	defer c.wg.Done()
+
+	// Create session to use when connecting to AWS Cloudwatch Logs
+	svc := c.sessionBuilder()
+
+	// Get events immediately when operator starts then use poll_interval duration.
+	c.getEvents(ctx, svc)
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-time.After(c.pollInterval.Duration):
+			c.getEvents(ctx, svc)
+		}
+	}
+}
+
+// sessionBuilder builds a session for AWS Cloudwatch Logs
+func (c *CloudwatchInput) sessionBuilder() *cloudwatchlogs.CloudWatchLogs {
+
+	region := aws.String(c.region)
+	var sess *session.Session
+	if c.profile != "" {
+		sess = session.Must(session.NewSessionWithOptions(session.Options{
+			Config: aws.Config{Region: region},
+
+			Profile: c.profile,
+		}))
+	}
+
+	if c.profile == "" {
+		sess = session.Must(session.NewSession(&aws.Config{
+			Region: region,
+		}))
+	}
+
+	return cloudwatchlogs.New(sess)
+}
```

Comment:
Would this work? If so, I think returning sooner than later is more consistent.

```suggestion
func (c *CloudwatchInput) sessionBuilder() *cloudwatchlogs.CloudWatchLogs {

	region := aws.String(c.region)
	var sess *session.Session
	if c.profile != "" {
		sess = session.Must(session.NewSessionWithOptions(session.Options{
			Config: aws.Config{Region: region},
			Profile: c.profile,
		}))
		return cloudwatchlogs.New(sess)
	}

	sess = session.Must(session.NewSession(&aws.Config{
	    Region: region,
	}))
	return cloudwatchlogs.New(sess)
}
```

Reply:
This does make sense. No reason to check again. Though I will switch the statement to look for a empty string instead of not an empty string.